palace/fem/libceed/basis.cpp

```diff
@@ -206,89 +158,59 @@ void InitMFEMInterpolatorBasis(const mfem::ParFiniteElementSpace &trial_fespace,
 }
 
 }  // namespace
 
-void InitBasis(const mfem::ParFiniteElementSpace &fespace, const mfem::FiniteElement &fe,
-               const mfem::IntegrationRule &ir, Ceed ceed, CeedBasis *basis)
+void InitBasis(const mfem::FiniteElement &fe, const mfem::IntegrationRule &ir,
+               CeedInt num_comp, Ceed ceed, CeedBasis *basis)
 {
-  // Check for fespace -> basis in hash table.
-  internal::BasisKey key(ceed, fespace, fe, ir);
-
-  // Initialize or retrieve key values (avoid simultaneous search and write).
-  auto basis_itr = internal::basis_map.end();
-  PalacePragmaOmp(critical(InitBasis))
+  if constexpr (false)
   {
-    basis_itr = internal::basis_map.find(key);
+    std::cout << "New basis (" << ceed << ", " << &fe << ", " << &ir << ")\n";
   }
-  if (basis_itr == internal::basis_map.end())
+  const bool tensor = dynamic_cast<const mfem::TensorBasisElement *>(&fe) != nullptr;
+  const bool vector = fe.GetRangeType() == mfem::FiniteElement::VECTOR;
+  if (tensor && !vector)
   {
-    const bool tensor = dynamic_cast<const mfem::TensorBasisElement *>(&fe) != nullptr;
-    const bool vector = fe.GetRangeType() == mfem::FiniteElement::VECTOR;
-    if (tensor && !vector)
-    {
-      InitTensorBasis(fespace, fe, ir, ceed, basis);
-    }
-    else
-    {
-      InitNonTensorBasis(fespace, fe, ir, ceed, basis);
-    }
-    PalacePragmaOmp(critical(InitBasis))
-    {
-      internal::basis_map[key] = *basis;
-    }
-    // std::cout << "New basis (" << ceed << ", " << &fe  << ", " << &ir << ")\n";
+    InitTensorBasis(fe, ir, num_comp, ceed, basis);
   }
   else
   {
-    *basis = basis_itr->second;
-    // std::cout << "Reusing basis (" << ceed << ", " << &fe  << ", " << &ir << ")\n";
+    InitNonTensorBasis(fe, ir, num_comp, ceed, basis);
   }
 }
 
-void InitInterpolatorBasis(const mfem::ParFiniteElementSpace &trial_fespace,
-                           const mfem::ParFiniteElementSpace &test_fespace,
-                           const mfem::FiniteElement &trial_fe,
-                           const mfem::FiniteElement &test_fe, Ceed ceed, CeedBasis *basis)
+void InitInterpolatorBasis(const mfem::FiniteElement &trial_fe,
+                           const mfem::FiniteElement &test_fe, CeedInt trial_num_comp,
+                           CeedInt test_num_comp, Ceed ceed, CeedBasis *basis)
 {
-  // Check for fespace -> basis in hash table.
-  internal::InterpBasisKey key(ceed, trial_fespace, test_fespace, trial_fe, test_fe);
-
-  // Initialize or retrieve key values (avoid simultaneous search and write).
-  auto basis_itr = internal::interp_basis_map.end();
-  PalacePragmaOmp(critical(InitInterpBasis))
+  if constexpr (false)
   {
-    basis_itr = internal::interp_basis_map.find(key);
+    std::cout << "New interpolator basis (" << ceed << ", " << &trial_fe << ", " << &test_fe
+              << ")\n";
   }
-  if (basis_itr == internal::interp_basis_map.end())
+  if constexpr (false)
```

Comment:
This is a remnant from the initial libCEED prolongation implementation which created the coarse to fine prolongation using interpolation at quadrature points instead of using MFEM's knowledge of the finite element space. It's better to just use `FiniteElement::GetTransferMatrix` from MFEM but I left the libCEED code around in case it becomes useful someday.